overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020-2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	SnapInstanceName          string        `json:"snap-instance-name"`
+	SnapRevision              snap.Revision `json:"snap-revision"`
+	ExportedForSnapdAsVersion string        `json:"exported-for-snapd-as-version,omitempty"`
+
+	// SourceIsHost is only true if provider of the manifest is not a snap but the classic system.
+	// All SourcePath fields, as visible through Sets[*].Exports[*].SourcePath, are absolute names
+	// in the host file system.
+	SourceIsHost bool `json:"source-is-host,omitempty"`
+
+	// Sets describe groups of exported files.
+	Sets map[string]ExportSet `json:"sets"`
+}
+
+// ExportSet describes a group of files for a given type of consumer.
+type ExportSet struct {
+	Name string `json:"name"`
+	// ConsumerIsHost is true if an export set provides files usable from the host system.
+	ConsumerIsHost bool                    `json:"consumer-is-host,omitempty"`
```

Comment:
we probably need to chat, there is something not anymore completely clear to me about why we decide this at this level

Reply:
I think we can remove this entirely from this intiial PR? AIUI this not needed for the initial MVP of "let's export the snapd tooling"?